### ISSUE TITLE
[importing libigl to other platform] c# wrapper or single dll can be used

### ISSUE THREAD

**Issue by xarthurx** [libigl/libigl#1418]
#### Describe your issue

Hi, I'm investigating the possibility of using libigl as a geometry processing core and import it into other languages so that I can write plugins for different software and use the geometry processing power from `libigl`. Currently C# within the .NET framework.

 With some reading, it seems to use cpp libraries inside a c# program or library, one need to compile the cpp library into a `dll` so that it can be imported.

My question is:
1. As `libigl` is a rather big lib with many plugins, would this be possible? Even for the basic functions? (not the plugins it supported).
2. Since `libigl` use Eigen intensively, how to also incorporating that if the above point is possible?
3. From this issue: https://github.com/libigl/libigl/issues/276 it seems there's no plan for providing a wrapper officially for c#, is it still true now? And does it mean neither of `point 1, 2` is valid?

I don't know much about programming language compilation... sorry if the question is too naive...

#### Check all that apply (change to `[x]`)
- [x] Windows
- [ ] Mac OS X
- [ ] Linux
- [ ] I have tried the `dev` branch and the problem persists

See https://libigl.github.io/CONTRIBUTING/#bugreport for more tips.

**Comment by jdumas** [comment]
Yeah nope, still not a big priority at the moment. By default we build a static lib, so you'd have to fiddle with the CMake yourself to build it as a .dll. Maybe we can make it an option if the changes are minimal, but I have zero experience with C# myself so you're on your own for this.

**Comment by lrgstu** [comment]
Another option is making a minimal C/C++ DLL that acts as a sort of proxy for other native libraries. Then you can P/Invoke the proxy in your C#. I sometimes do this so I can control the API (which includes the types of data I need to marshal). A smaller benefit is being able to use different styles of libraries in the proxy without fuss (dynamic/static/header-only).

Also, I read another post of someone trying to do this DLL business with CLR. AFAIK there is no reason to involve CLR in this. In fact I would avoid it because it isn't portable.
